star-dash/star-dash/Achievement/AchievementManager.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  AchievementManager.swift
+//  star-dash
+//
+//  Created by Ho Jun Hao on 31/3/24.
+//
+
+import Foundation
+
+class AchievementManager: EventListener {
+    static let shared = AchievementManager()
+
```

Comment:
in my Peggle, the comment was that singleton pattern exposes it to everyone in the project, instead we should be using dependency injection to limit access to specific classes

Reply:
removed singleton design pattern